src/location.rs

```diff
@@ -1,10 +1,43 @@
 use bevy::prelude::*;
+use crate::direction::{Direction};
 
-pub type XCoord = u8;
-pub type YCoord = u8;
+type Coord = u8;
 
-#[derive(Component, Default, Debug)]
+const MAX_X: Coord = 15;
+const MAX_Y: Coord = 8;
+
+#[derive(Component, Debug)]
 pub struct Location {
-    x: XCoord,
-    y: YCoord,
+    x: Coord,
+    y: Coord,
+}
+
+impl Location {
+    pub fn new() -> Self {
+        Self { x: MAX_X / 2, y: MAX_Y / 2 }
+    }
+
+    pub fn move_by_one(&mut self, dir: &Direction) -> Result<(), ()> {
```

Comment:
Also to rust style, I would normally prefer a functional-style `fn adjacent(&self, dir: &Direction) -> Result<Location, ()>`. Also, I would put the out-of-bounds check in another system so all the Option aspect of this could be factored out you could just have `fn adjacent(&self, &Direction) -> Location` especially if we used signed coords and guaranteed not to overflow.